### PR TITLE
Feat/bugfix

### PR DESCRIPTION
**Pull request by Leon-pix-coder**
Feat/bugfix

### CODE REVIEW

File: src/components/VoteItem/index.tsx [hunk 1]
```diff
@@ -108,9 +111,12 @@ const VoteItem = ({
       shapes: [HEART_SHAPE],
       zIndex: 10,
     });
-  }
+  };
 
-  const onVoteClick = () => {
+  const onVoteClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    console.log(event);
```
Comment (evan-spidy) [inline src/components/VoteItem/index.tsx]: Remove console log

Comment (Leon-aelf-coder) [reply src/components/VoteItem/index.tsx]: removed

File: src/components/ForYou/index.tsx [hunk 2]
```diff
@@ -103,6 +111,22 @@ const ForYou = ({
...
   };
 
+  const { run: chooseIntrest } = useThrottleFn(
+    async () => {
+      setShowChoosen(false);
+      updateUserStatus(false);
+      try {
+        await postWithToken("/api/app/discover/choose", {
+          chainId,
+          choices: intrests,
```
Comment (evan-spidy) [inline src/components/ForYou/index.tsx]: wrong spelling

File: src/components/ForYou/index.tsx [hunk 3]
```diff
@@ -129,7 +153,7 @@ const ForYou = ({
                 className="h-screen flex flex-col relative items-center"
               >
                 {item.appType === "AD" && index === currentIndex ? (
```
Comment (evan-spidy) [inline src/components/ForYou/index.tsx]: make 'AD' as a variable

File: src/components/ForYou/index.tsx [hunk 4]
```diff
@@ -154,17 +178,26 @@ const ForYou = ({
...
             Select Your Areas of Interest
           </span>
           <span className="block mt-[8px] mb-[24px] text-[9px] leading-[10px]">
             Your preferences will help us create a journey unique to you.
           </span>
 
-          <CheckboxGroup options={DISCOVER_CATEGORY} onChange={console.log} />
+          <CheckboxGroup
+            options={DISCOVER_CATEGORY.slice(1)}
+            onChange={setIntrests}
```
Comment (evan-spidy) [inline src/components/ForYou/index.tsx]: same here

File: src/components/Home/index.tsx [hunk 5]
```diff
@@ -193,7 +192,10 @@ const Home = ({ onAppItemClick, recommendList }: IHomeProps) => {
                 </span>
               </div>
               <div className="col-6 h-[230px] flex flex-col gap-[10px]">
-                <div className="col-12 p-[13px] flex-1 bg-tertiary rounded-[18px]">
+                <div
+                  className="col-12 p-[13px] flex-1 bg-tertiary rounded-[18px]"
+                  onClick={() => onAppItemClick()}
```
Comment (evan-spidy) [inline src/components/Home/index.tsx:197]: can just use onclick={onAppItemClick}

File: src/components/InviteFriends/index.tsx [hunk 6]
```diff
@@ -25,77 +25,93 @@ const InviteFriendsStatus = ({
...
+        body: new URLSearchParams(requestObject).toString(),
+        headers: {
+          "Content-Type": "application/x-www-form-urlencoded",
+        },
+      });
+      const portKeyResObj = await portKeyRes.json();
+      if (portKeyRes?.ok && portKeyResObj?.access_token) {
+        const token = portKeyResObj.access_token;
+        const res = await fetch(
+          portkeyServer +
```
Comment (evan-spidy) [inline src/components/InviteFriends/index.tsx:80]: full cap for constants

File: src/components/TaskModule/components/TaskItem/index.tsx [hunk 7]
```diff
@@ -161,15 +161,31 @@ const TaskItem = ({
...
       );
+      if (jumpItem) {
+        const isComplete = await openNewPageWaitPageVisible(
+          jumpItem.url,
+          taskId,
+          () =>
+            fetchWithToken(
+              `/api/app/user/complete-task?${new URLSearchParams({
+                chainId,
+                userTask: userTask,
```
Comment (evan-spidy) [inline src/components/TaskModule/components/TaskItem/index.tsx]: if it is the same name you do not need to pass in like this, cna just userTask

File: src/components/TaskModule/utils/index.ts [hunk 8]
```diff
@@ -13,24 +16,34 @@ export const openNewPageWaitPageVisible = (
...
-      onFinish?.();
+      return req()
+        .then(() => {
+          window?.Telegram?.WebApp?.openTelegramLink?.(url);
+          return true;
+        })
+        .catch(() => {
+          window?.Telegram?.WebApp?.openTelegramLink?.(url);
+          return false;
+        });
```
Comment (evan-spidy) [inline src/components/TaskModule/utils/index.ts:27]: what are you trying to do?

File: src/components/VoteItem/index.tsx [hunk 9]
```diff
@@ -69,11 +74,14 @@ const VoteItem = ({
         setLoading(true);
         const { data } = await voteRequest(rawTransaction, result);
         if (data.status === VOTE_STATUS.VOTED) {
```
Comment (evan-spidy) [inline src/components/VoteItem/index.tsx]: you should check data here also

File: src/components/TaskModule/utils/index.ts (hunk 8 above)
Comment (Leon-pix-coder) [reply src/components/TaskModule/utils/index.ts:27]: This code comes from tomorrowDao, Req calls the interface to complete the task. If it succeeds, it returns true, if it fails, it returns false.

File: src/components/Home/index.tsx (hunk 5 above)
Comment (Leon-pix-coder) [reply src/components/Home/index.tsx:197]: onAppItemClick has params type limit

File: src/components/ForYou/index.tsx (hunk 3 above)
Comment (Leon-pix-coder) [reply src/components/ForYou/index.tsx]: Updated

File: src/components/InviteFriends/index.tsx (hunk 6 above)
Comment (Leon-pix-coder) [reply src/components/InviteFriends/index.tsx:80]: Updated

File: src/components/TaskModule/components/TaskItem/index.tsx (hunk 7 above)
Comment (Leon-pix-coder) [reply src/components/TaskModule/components/TaskItem/index.tsx]: Updated

File: src/components/VoteItem/index.tsx (hunk 9 above)
Comment (Leon-pix-coder) [reply src/components/VoteItem/index.tsx]: Updated

File: src/components/ForYou/index.tsx (hunk 4 above)
Comment (Leon-pix-coder) [reply src/components/ForYou/index.tsx]: Updated

File: src/components/ForYou/index.tsx (hunk 2 above)
Comment (Leon-pix-coder) [reply src/components/ForYou/index.tsx]: Updated

**Pull request merged**